generate_buddy_headers.py

```diff
@@ -132,33 +157,34 @@ def generate_header_file(yaml_file_name, header_file_name, mmu, list, includes):
             err_code = int(code)
             assert err_code not in err_dict, f"Duplicate error code {code}."
 
-            if mmu:
+            btns = []
+            if "action" in err:
                 btns = [f"ButtonOperations::{action}" for action in err["action"]]
 
-                # Make sure the array always has 3 buttons, NoOperation button
-                # occupying the appropriate "empty" slots
-                if len(btns) == 0:
-                    btns.append("ButtonOperations::NoOperation")
-                if len(btns) == 1:
-                    btns.insert(0, "ButtonOperations::NoOperation")
-                if len(btns) == 2:
-                    btns.append("ButtonOperations::NoOperation")
+            # Make sure the array always has 3 buttons, NoOperation button
+            # occupying the appropriate "empty" slots
+            if len(btns) == 0:
+                btns.append("ButtonOperations::NoOperation")
+            if len(btns) == 1:
+                btns.insert(0, "ButtonOperations::NoOperation")
+            if len(btns) == 2:
+                btns.append("ButtonOperations::NoOperation")
```

Comment:
This three button "layout" is specific to MMU, I think generic dialogs in the printer can have up to 4 buttons, and they are not spaced this way, they are just placed at the beginning of an array of length 4 and the rest is filled up with some `none`, i.e. `ButtonOperations::NoOperation` in this context.

Reply:
force-pushed